pytestqt/qt_compat.py

```diff
@@ -109,16 +124,38 @@ def get_versions():
             QWidget = _QtWidgets.QWidget
             qInstallMessageHandler = QtCore.qInstallMessageHandler
             qt_api_name = 'PyQt5'
+
+            QStandardItem = QtGui.QStandardItem
+            QStandardItemModel = QtGui.QStandardItemModel
+            QFileSystemModel = _QtWidgets.QFileSystemModel
+            QStringListModel = QtCore.QStringListModel
+            QSortFilterProxyModel = QtCore.QSortFilterProxyModel
+
         else:
             QApplication = QtGui.QApplication
             QWidget = QtGui.QWidget
             qInstallMsgHandler = QtCore.qInstallMsgHandler
             qt_api_name = 'PyQt4'
 
+            QStandardItem = QtGui.QStandardItem
+            QStandardItemModel = QtGui.QStandardItemModel
+            QFileSystemModel = QtGui.QFileSystemModel
+            QStringListModel = QtGui.QStringListModel
+            QSortFilterProxyModel = QtGui.QSortFilterProxyModel
+
+
         def get_versions():
             return VersionTuple(qt_api_name, QtCore.PYQT_VERSION_STR,
                                 QtCore.qVersion(), QtCore.QT_VERSION_STR)
 
+        def cast(obj, typ):
+            """casts from a subclass to a parent class"""
+            return sip.cast(obj, typ)
+
+        def extract_from_variant(variant):
+            """returns python object from the given QVariant"""
+            return variant.toPyObject() if variant is not None else None

```

Comment:
This looks wrong - shouldn't this be in the PySide section? I don't know about PyQt4, but it's definitely the wrong thing to do for PyQt4.

I think there should be a test for this too - maybe something like this (untested):

```python
def test_qvariant(tmpdir):
    settings = QtCore.QSettings(str(tmpdir / 'foo.ini'), QSettings.IniFormat)
    if ...:
        settings.setValue("normal", QtCore.QVariant(42))
        settings.setValue("empty", QtCore.QVariant())
    else:
        settings.setValue("normal", 42)
        settings.setValue("empty", None)

    ret_normal = settings.value("normal")
    ret_empty = settings.value("empty")

    assert qt_compat.extract_from_variant(ret_normal) == 42
    assert qt_compat.extract_from_variant(ret_empty) is None
```